### PR TITLE
magnets.js

### PR DESCRIPTION
**Pull request by SetupandDraw**
I've turned the attractor object into a magnet object with a boolean property to make it a repeller or an attractor
depending if you press a key when clicking on the canvas or not, maybe this could be useful for someone.
This is a one js file only alternative version; containing constructors functions for particles and magnets.

### CODE REVIEW

File: challenges/CC_56_attraction_repulsion/magnets.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    magnets.push(new Magnet(mouseX, mouseY, true));
+  } else {
+    magnets.push(new Magnet(mouseX, mouseY, false));
+  }
+}
+
+// here's the constructor function for the magnets
+function Magnet(x, y, t) {
+    this.pos = createVector(x, y);
+    this.repeller = t; // type of attractor
```
Comment (SetupandDraw) [inline challenges/CC_56_attraction_repulsion/magnets.js]: In the comment I meant type of magnet, it could be an attractor or a repeller

**Comment by shiffman** [comment]
Hey this is fantastic! This repo, however, is meant to have the code that matches the video exactly so others can start from that code and make their own improvements. I was thinking of putting READMEs in the challenge folders then people could link out to their improved code. Would you like to pull request that?

**Comment by SetupandDraw** [comment]
Sorry, I am still new to GitHub :( 

Sure I would like to contribute more! I know you have a couple of good tutorials on GitHub, I will dig into that in the next days and better understand how the whole Git thing works.

Have a great day!

Luca

Il giorno 22/feb/2017, alle ore 15:07, Daniel Shiffman <notifications@github.com> ha scritto:

**Comment by shiffman** [comment]
No need to apologize, we are all learning how to do this here!  My suggestion is simply to make your own repo with your version and then I'll create a link to it here within a README.md file.  You can see an example here.  No rush and let me know how I can help!

https://github.com/tadimon/Rainbow-Code/tree/master/challenges/CC_58_EarthQuakeViz3D

**Comment by SetupandDraw** [comment]
I will, Thank you Daniel!

Lu

Il giorno 23/feb/2017, alle ore 02:36, Daniel Shiffman <notifications@github.com> ha scritto:

**Comment by SetupandDraw** [comment]
Ciao Daniel,

I've finally made a repo on GitHub with my code for this. Please have a look at it here: https://github.com/SetupandDraw/Learning-various-things-in-javascript/tree/master/Magnets
if that looks good to you and you think it could be useful for your students, please include a link to it. If I missed something, or if you think  some changes are needed, just let me know.

Looking forward to upload something on Fractal Spirograph as well, I will keep you posted if interested.

Have a great day!

Luca

Il giorno 23/feb/2017, alle ore 02:36, Daniel Shiffman <notifications@github.com> ha scritto:

**Comment by shiffman** [comment]
Great! I'll close this pull request, can you make a new one with a link to your repo in a README.md file in `challenges/CC_56_attraction_repulsion/magnets.js`?

**Comment by SetupandDraw** [comment]
Sure, just made it!

Have a great day!

**Comment by SetupandDraw** [comment]
Hi there!

Don't know if I have done something wrong with my pull request attempt or if you simply didn't have time for this yet.

Please let me know If I have to adjust something in my code/GithHub-thing.

Have a great day!

Lu

 
Il giorno 06/mar/2017, alle ore 10:25, Luca Marchettoni <lucamarchettoni@gmail.com> ha scritto:

**Comment by shiffman** [comment]
Hi, apologies! I don't see the new pull request adding a link to your repo from a `README.md`? Can you link to it? I can also just make the README if you prefer!

**Pull request closed without merging**